src/clincoded/static/components/globals.js

```diff
@@ -152,13 +152,15 @@ export const addQueryKey = function(href, key, value, allowEmptyValue=true) {
  * This method allows you to add multiple queries of the same key.
  * 
  * @param {string} href - the base url the queries will add to
- * @param {Array<{ key: string, value: string }>} keyValuePairs - an array of key value pairs that will be added as queries to `href`
+ * @param {Array<{ key: string, value: string } | Array<string>>} keyValuePairs - an array of key value pairs that will be added as queries to `href`
  * @param {boolean} allowEmptyValue - whether to add empty-valued query to url; if `false`, will ignore queries whose value is empty string, null or undefined. This argument will just be passed to `addQueryKey()`.
  * @returns {string} the url embedded with queries
  */
 module.exports.getQueryUrl = function(href, keyValuePairs=[], allowEmptyValue=true) {
     return keyValuePairs.reduce((accumulatedUrl, currentPair) => {
-        return addQueryKey(accumulatedUrl, currentPair.key, currentPair.value, allowEmptyValue);
+        return Array.isArray(currentPair) ? 
+            addQueryKey(accumulatedUrl, currentPair[0], currentPair[1], allowEmptyValue) :
+            addQueryKey(accumulatedUrl, currentPair.key, currentPair.value, allowEmptyValue);
```

Comment:
I think it's confusing that keyValuePairs can contain an array or keyValuePair object.  In your call to getQureyUrl, it's easy to just pass in array of keyValuePair object; then no need to change this function.  May be there's reason to use array?

Reply:
I agree that we can just use an object of key: value: for the query params. The reason to add tuple-style array is because I found over time it's a bit tedious to write both key: and value: alongside the actual values, especially when we have many params.

Compare the tuples
```js
[
  ['content-type', 'application/json'],
  ['hgvs', '1'],
  ['protein', '1'],
  ['xref_refseq', '1'],
  ['ExAC', '1'],
  ['MaxEntScan', '1'],
  ['GeneSplicer', '1'],
  ['Conservation', '1'],
  ['numbers', '1'],
  ...
]
```

to key: - value: objects:

```js
[
  { key: 'content-type', value: 'application/json' },
  { key: 'hgvs', value: '1' },
  { key: 'protein', value: '1' },
  { key: 'xref_refseq', value: '1' },
  { key: 'ExAC', value: '1' },
  { key: 'MaxEntScan', value: '1' },
  { key: 'GeneSplicer', value: '1' },
  { key: 'Conservation', value: '1' },
  { key: 'numbers', value: '1' },
  ...
]
```

But the trade off I think is like you said, using key: value: would be more explicit and avoid confusion, versus, using array as tuples to quickly specify key value pairs. I'm thinking we can choose either one to stick to - which one would you think is better?

Reply:
It doesn't matter to me which one actually as long as only one format is used.  Thanks.

Reply:
Alright, Iooks like it's easier to just fallback to using key: value: since there're other places using it.